samples/esb/esb_monitor/src/sniffer.c

```diff
@@ -0,0 +1,159 @@
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#if defined(CONFIG_ESB_SNIFFER)
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+
+#include <hal/nrf_radio.h>
+#include <esb.h>
+
+#include <SEGGER_RTT.h>
+
+#include "sniffer.h"
+
+LOG_MODULE_REGISTER(esb_sniffer, CONFIG_ESB_MONITOR_APP_LOG_LEVEL);
+
+static uint8_t rtt_data_buff[CONFIG_ESB_SNIFFER_RTT_DATA_BUFF_SZ];
+static uint8_t rtt_comm_ret_buff[CONFIG_ESB_SNIFFER_RTT_COMM_RET_BUFF_SZ];
+static uint8_t rtt_comm_buff[CONFIG_ESB_SNIFFER_RTT_COMM_BUFF_SZ];
+
+static k_tid_t rtt_comm_thread_id;
+static K_THREAD_STACK_DEFINE(rtt_comm_thread_stack, CONFIG_ESB_SNIFFER_THREAD_STACK_SZ);
+static struct k_thread rtt_comm_thread;
+
+static struct esb_sniffer_cfg sniffer_cfg;
+enum esb_sniffer_rtt_comm {
+	SNIFFER_START  = 1,
+	SNIFFER_STOP   = 2,
+	SNIFFER_STATUS = 3
+};
+
+static void radio_reg_to_array(uint32_t addr, char *array)
+{
+#if __CORTEX_M == (0x04U)
+	uint32_t inp = addr;
+
+	*((uint32_t *)array) = sys_cpu_to_be32((uint32_t)__RBIT(inp));
+#else
+	uint32_t inp = sys_cpu_to_le32(addr);
+
+	inp = (inp & 0xF0F0F0F0) >> 4 | (inp & 0x0F0F0F0F) << 4;
+	inp = (inp & 0xCCCCCCCC) >> 2 | (inp & 0x33333333) << 2;
+	inp = (inp & 0xAAAAAAAA) >> 1 | (inp & 0x55555555) << 1;
+	*((uint32_t *)array) =  inp;
+#endif
+}
+
+static void rtt_comm_thread_fn(void)
+{
+	enum esb_sniffer_rtt_comm comm;
+	uint32_t read_len;
+	int err;
+
+	/* Send packet length to backend */
+	read_len = sizeof(struct rtt_frame);
+	read_len = sys_cpu_to_be32(read_len);
+
+	SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+						&read_len, sizeof(uint32_t));
+	while (1) {
+		read_len = SEGGER_RTT_Read(CONFIG_ESB_SNIFFER_RTT_COMM_CHANNEL,
+						&comm, sizeof(enum esb_sniffer_rtt_comm));
+		if (read_len == sizeof(enum esb_sniffer_rtt_comm)) {
+			switch (comm) {
+			case SNIFFER_START:
+				LOG_INF("Start sniffer");
+				if (!sniffer_cfg.is_running) {
+					err = esb_start_rx();
+				} else {
+					err = 0;
+				}
+
+				if (err != 0) {
+					LOG_ERR("Failed to start sniffer, err: %d", err);
+				} else {
+					sniffer_cfg.is_running = true;
+				}
+
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			case SNIFFER_STOP:
+				LOG_INF("Stop sniffer");
+				if (sniffer_cfg.is_running) {
+					err = esb_stop_rx();
+				} else {
+					err = 0;
+				}
+
+				if (err != 0) {
+					LOG_ERR("Failed to stop sniffer, err: %d", err);
+				} else {
+					sniffer_cfg.is_running = false;
+				}
+
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			case SNIFFER_STATUS:
+				err = sniffer_cfg.is_running;
+				SEGGER_RTT_Write(CONFIG_ESB_SNIFFER_RTT_COMM_RET_CHANNEL,
+										&err, sizeof(int));
+				break;
+			default:
+				LOG_ERR("Unrecognized rtt command: %d", comm);
+			}
+		}
+
+		k_sleep(K_MSEC(500));
+	}
```

Comment:
Too deep, please break up into functions, or at least inverse the if logic to avoid this unnecessary depth.